libdevcrypto/Common.cpp

```diff
@@ -275,8 +275,8 @@ bytesSec dev::scrypt(std::string const& _pass, bytes const& _salt, uint64_t _n,
 void KeyPair::populateFromSecret(Secret const& _sec)
 {
 	m_secret = _sec;
-	if (s_secp256k1pp.verifySecret(m_secret, m_public))
-		m_address = toAddress(m_public);
+	m_public = toPublic(_sec);
```

Comment:
The check is done in toPublic anyway, it will return zero in case of invalid key. As far as I understand crypto++ validate() checks for curve validity, which is not needed for libsecp256k1 anyway and also checks that the point is valid on the curve, which secp256k1_ec_pubkey_create also does.
The previous behavior of this function for invalid secret was to set m_secret to invalid key, but leave m_public and m_address to the previous value. With this change m_public and m_address will be set to zero, which is better since these don't throw anyway.